ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionValidator.java

```diff
@@ -112,7 +112,8 @@ public MainnetTransactionValidator(
       return signatureResult;
     }
 
-    if (goQuorumCompatibilityMode && !transaction.getGasPrice().isZero()) {
+    if (goQuorumCompatibilityMode
+        && !transaction.getGasPrice().map(Wei::isZero).orElse(Boolean.TRUE)) {
```

Comment:
This clause is getting harder and harder to understand. Maybe we could encapsulate the "tx cost zero" logic into a method in the Transaction class?

Something like: `transaction.hasGasPrice()` or `transaction.isGasPriceZero()` (I'm not great with names, feel free to come up with a better name for this method).

I recon this would make this if statement easier to read:

```
if (goQuorumCompatibilityMode && transaction.hasGasPrice()) {
    ....
}
```

Reply:
encapsulated in core Transaction.  I went ahead and added 1559 fields just in case.